### PR TITLE
feat: add bootnode cli command

### LINKED ISSUE #5872: Lodestar Bootnode Functionality
### Problem description

Currently, we do not have an easy method for Infrastructure to setup bootnodes for our use. Our images do not support this functionality and there is nothing implemented to ensure we have the best security/networking practices similar to how other clients are enabling their own bootnodes on the network.

### Solution description

We should investigate and implement a solution that will allow us to easily setup and run reliable bootnodes for the network. I imagine something as simple as `./lodestar bootnode {flags}` running our docker image. 

This would allow Infrastructure and other community members to easily setup a bootnode for use with our discv5 implementation and allow us to fulfill our promise to the community by diversifying bootnode infrastructure for the network as mentioned here: https://ethresear.ch/t/execution-consensus-client-bootnodes/14588/42

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wemeetagain**
**Motivation**

- Resolve #5872 

**Description**

- Add a `lodestar bootnode` cli command

### CODE REVIEW

File: packages/cli/src/cmds/bootnode/handler.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    abortController.signal.addEventListener(
+      "abort",
+      async () => {
+        try {
+          await metricsServer?.close();
+          await discv5.stop();
+          logger.debug("Bootnode closed");
+          // Explicitly exit until active handles issue is resolved
+          // See https://github.com/ChainSafe/lodestar/issues/5642
+          process.exit(0);
```
Comment (nflaig) [inline packages/cli/src/cmds/bootnode/handler.ts]: explicit exit should not be required for bootnode, issue is related to libp2p

File: packages/cli/src/cmds/bootnode/options.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  bootnodesFile?: string;
+  persistNetworkIdentity?: boolean;
+  "enr.ip"?: string;
+  "enr.ip6"?: string;
+  "enr.udp"?: number;
+  "enr.udp6"?: number;
+  nat?: boolean;
+};
+
+export const bootnodeExtraOptions: CliCommandOptions<BootnodeExtraArgs> = {
```
Comment (nflaig) [inline packages/cli/src/cmds/bootnode/options.ts:22]: might be useful to support `--bootnodes` flag as well

**Review by nflaig — APPROVED** [review]
LGTM!

File: packages/cli/src/cmds/bootnode/handler.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      }
+      logger.info("Server metrics", {
+        connectedPeers: discv5.connectedPeerCount,
+        activeSessions: discv5.sessionService.sessionsSize(),
+        ip4Nodes: ip4Only,
+        ip6Nodes: ip6Only,
+        ip4AndIp6Nodes: ip4ip6,
+        unreachableNodes: unreachable,
+      });
+    }, 10_000);
```
Comment (dapplion) [inline packages/cli/src/cmds/bootnode/handler.ts:115]: Metrics should be done with prometheus as first medium, dumping to logs becomes noise for processes expected to run for days or weeks

Comment (wemeetagain) [reply packages/cli/src/cmds/bootnode/handler.ts:115]: Agree that metrics should be done with prometheus first. I think some logging can be useful tho.

I used this as reference: https://github.com/sigp/lighthouse/blob/dfcb3363c757671eb19d5f8e519b4b94ac74677a/boot_node/src/server.rs#L78-L121

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.11.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.11.0) :tada:

**Pull request merged**